web3.md

```diff
@@ -1439,17 +1447,36 @@ Collecting Coverage Data
+    rule #parseByteCodeAux(PCOUNT, SIZE, PGM, SCHED, OPLIST) => #parseByteCodeAux(PCOUNT +Int #widthOp(#dasmOpCode(PGM [ PCOUNT ], SCHED)), SIZE, PGM, SCHED, OPLIST ListItem({PCOUNT | #dasmOpCode(PGM [ PCOUNT ], SCHED) | #unparseDataByteArray(substrBytes(PGM, PCOUNT +Int 1, PCOUNT +Int #widthOp(#dasmOpCode(PGM [ PCOUNT ], SCHED))))}))
       requires PCOUNT <Int SIZE
 
-    rule <k> #execute ... </k>
-         <pc> PCOUNT </pc>
-         <execPhase> EPHASE </execPhase>
-         <program> PGM </program>
-         <opcodeCoverage> ... { keccak(PGM) | EPHASE } |-> (PCS (.Set => SetItem(PCOUNT))) ... </opcodeCoverage>
-      requires notBool PCOUNT in PCS
+    rule <k> #gas [ OP , AOP ] ... </k>
+         <pc>             PCOUNT                                                                </pc>
+         <previousPC>     _ => PCOUNT                                                           </previousPC>
+         <programID>      ({HASH|EPHASE} #as PGMID):ProgramIdentifier                           </programID>
+         <opcodeCoverage> ... PGMID |-> (PCS => PCS (PCOUNT |-> ListItem(#getWSArgs(OP,AOP)))) ... </opcodeCoverage>
+      requires notBool PCOUNT in_keys(PCS)
+      [priority(25)]
+
+    rule <k> #gas [ OP , AOP ] ... </k>
+         <pc>             PCOUNT                                                                                    </pc>
+         <previousPC>     PREV => PCOUNT                                                                            </previousPC>
+         <programID>      ({HASH|EPHASE} #as PGMID):ProgramIdentifier                                               </programID>
+         <opcodeCoverage> ... PGMID |-> (PCS => PCS[PCOUNT <- ({PCS[PCOUNT]}:>List ListItem(#getWSArgs(OP,AOP)))]) ... </opcodeCoverage>
+      requires PREV =/=Int PCOUNT
+       andBool PCOUNT in_keys(PCS)
       [priority(25)]
 
+    syntax List ::= #getWSArgs ( OpCode, OpCode ) [function]
+ // --------------------------------------------------------
+    rule #getWSArgs (QOP:CallOp     , CO  W0 W1 W2 W3 W4 W5 W6) => ListItem(W0) ListItem(W1) ListItem(W2) ListItem(W3) ListItem(W4) ListItem(W5) ListItem(W6)
+    rule #getWSArgs (CSO:CallSixOp  , CSO W0 W1 W2 W3 W4 W5   ) => ListItem(W0) ListItem(W1) ListItem(W2) ListItem(W3) ListItem(W4) ListItem(W5)
+    rule #getWSArgs (QOP:QuadStackOp, QOP W0 W1 W2 W3         ) => ListItem(W0) ListItem(W1) ListItem(W2) ListItem(W3)
+    rule #getWSArgs (TOP:TernStackOp, TOP W0 W1 W2            ) => ListItem(W0) ListItem(W1) ListItem(W2)
+    rule #getWSArgs (BOP:BinStackOp , BOP W0 W1               ) => ListItem(W0) ListItem(W1)
+    rule #getWSArgs (UOP:UnStackOp  , UOP W0                  ) => ListItem(W0)
+    rule #getWSArgs (OP             , OP                      ) => .List [owise]
```

Comment:
```suggestion
    rule #getWSArgs (OP:StackOp             , OP STACK                     ) => WordStack2List(STACK) [owise]
```

Probably already have some function like this.

Then an owise rule for everything else.